spec/lib/pdf_filler/nj1040_pdf_spec.rb

```diff
@@ -197,5 +208,53 @@
         expect(pdf_fields["ZIP Code"]).to eq "08037"
       end
     end
+
+    describe "filing status" do
+      context "single" do
+        before do
+          submission.data_source.direct_file_data.filing_status = 1
+        end
+        it "checks the appropriate box" do
+          expect(pdf_fields["Group1"]).to eq "Choice1"
+        end
+      end
+
+      context "married filing jointly" do
+        before do
+          submission.data_source.direct_file_data.filing_status = 2
+        end
+        it "checks the appropriate box" do
+          expect(pdf_fields["Group1"]).to eq "Choice2"
+        end
+      end
+
+      context "married filing separately" do
+        before do
+          submission.data_source.direct_file_data.filing_status = 3
+        end
+        it "checks the appropriate box" do
+          expect(pdf_fields["Group1"]).to eq "Choice3"
+        end
+      end
+
+      context "head of household" do
+        before do
+          submission.data_source.direct_file_data.filing_status = 4
+        end
+        it "checks the appropriate box" do
+          expect(pdf_fields["Group1"]).to eq "Choice4"
+        end
+      end
+
+      context "qualifying widow" do
+        before do
+          submission.data_source.direct_file_data.filing_status = 5
+          submission.data_source.direct_file_data.spouse_date_of_death = "2023-09-30"
```

Comment:
[pebble] Why this date as a magic string? Could we add a variable name for this to be more self-documenting? For example:
```
date_within_the_prior_year = "2023-09-30"
submission.data_source.direct_file_data.spouse_date_of_death = date_within_the_prior_year
```